streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -502,9 +502,8 @@ public <T> void addStoreLevelMutableMetric(final String taskId,
             storeLevelTagMap(taskId, metricsScope, storeName)
         );
         if (metrics.metric(metricName) == null) {
-            final MetricConfig metricConfig = new MetricConfig().recordLevel(recordingLevel);
+            metrics.addMetricIfAbsent(metricName, new MetricConfig().recordLevel(recordingLevel), valueProvider);
```

Comment:
Just take here as an example: here we depend on two threads would not try to create the same store-name and the same task-id but that may not always hold true in the future, so better using the newly introduced API. cc @cadonna .

Reply:
Agreed!